File: 2/a.cpp
#include "utils.h"

long long solve(vector<string> &lines) {
  long long result = 0;

  for (string &line : lines) {
    stringstream ss(line);
    char abc, xyz;
    ss >> abc >> xyz;
    const int a = abc - 'A', x = xyz - 'X';
    result += 3 * ((4 + x - a) % 3) + 1 + x;
  }

  return result;
}

File: 2/a.nim
import strutils, utils

var result = 0
for line in load().splitLines:
  let a = int(line[0]) - int('A')
  let x = int(line[2]) - int('X')
  result += 3 * ((4 + x - a) mod 3) + 1 + x
echo result